icu4c/source/test/cintltst/cloctst.c

```diff
@@ -6741,3 +6742,11 @@ static void TestBug20149() {
         log_err("ERROR: in uloc_forLanguageTag %s return %s\n", myErrorName(status), locale);
     }
 }
+
+// For the C locale, ICU should have a default locale of en_US_POSIX
```

Comment:
Please mention the previous ticket number here (ICU-20575) as a breadcumb for others.

Reply:
Please also add some comments that this test checks if the environment variable LANG is set, and if so ensures that C/C.UTF-8 cause ICU's default locale to be `en_US_POSIX`.